streams/src/test/java/org/apache/kafka/streams/integration/SuppressionIntegrationTest.java

```diff
@@ -313,6 +316,124 @@ public void shouldShutdownWhenBytesConstraintIsViolated() throws InterruptedExce
+            final boolean suppressedRecords = waitForAnyRecord(outputSuppressed);
+            final Properties config = CLUSTER.getLogConfig(changeLog);
+
+            assertThat(config.getProperty("retention.ms"), is(logConfig.get("retention.ms")));
+            assertThat(CLUSTER.getAllTopicsInCluster().contains(changeLog), is(true));
+            assertThat(rawRecords, Matchers.is(true));
+            assertThat(suppressedRecords, is(true));
+        } finally {
+            driver.close();
+            cleanStateAfterTest(CLUSTER, driver);
+        }
+    }
+
+    @Test
+    public void shouldWithLoggingDisable() {
+        final String testId = "-shouldWithLoggingDisable";
+        final String appId = getClass().getSimpleName().toLowerCase(Locale.getDefault()) + testId;
+        final String input = "input" + testId;
+        final String outputSuppressed = "output-suppressed" + testId;
+        final String outputRaw = "output-raw" + testId;
+        final String changeLog = "suppressionintegrationtest-shouldWithLoggingDisable-KTABLE-SUPPRESS-STATE-STORE-0000000004-changelog";
+
+        cleanStateBeforeTest(CLUSTER, input, outputRaw, outputSuppressed);
+
+        final StreamsBuilder builder = new StreamsBuilder();
+
+        final KStream<String, String> inputStream = builder.stream(input);
+
+        final KTable<String, String> valueCounts = inputStream
+            .groupByKey()
+            .aggregate(() -> "()", (key, value, aggregate) -> aggregate + ",(" + key + ": " + value + ")");
+
+        valueCounts
+            .suppress(untilTimeLimit(ofMillis(MAX_VALUE), maxRecords(1L)
+                .emitEarlyWhenFull()
+                .withLoggingEnabled(Collections.emptyMap())))
+            .toStream()
+            .to(outputSuppressed);
+
+        valueCounts
+            .toStream()
+            .to(outputRaw);
+
+        final Properties streamsConfig = getStreamsConfig(appId);
+        streamsConfig.put(StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.StringSerde.class);
+        streamsConfig.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, Serdes.StringSerde.class);
+
+        final KafkaStreams driver = IntegrationTestUtils.getStartedStreams(streamsConfig, builder, true);
+        try {
+            produceSynchronously(
+                input,
+                asList(
+                    new KeyValueTimestamp<>("k1", "v1", scaledTime(0L)),
+                    new KeyValueTimestamp<>("k1", "v2", scaledTime(1L)),
+                    new KeyValueTimestamp<>("k2", "v1", scaledTime(2L)),
+                    new KeyValueTimestamp<>("x", "x", scaledTime(3L))
+                )
+            );
+            final boolean rawRecords = waitForAnyRecord(outputRaw);
+            final boolean suppressedRecords = waitForAnyRecord(outputSuppressed);
+            final Properties config = CLUSTER.getLogConfig(changeLog);
+
+            assertThat(config.getProperty("retention.ms"), Matchers.is(nullValue()));
```

Comment:
```suggestion
            assertThat(CLUSTER.getAllTopicsInCluster(), contains(changeLog));
```

And we can remove line 426, `final Properties config = CLUSTER.getLogConfig(changeLog);`, since it would be unused.